src/App.tsx

```diff
@@ -0,0 +1,32 @@
+import './App.css';
+import Loading from './components/Loading/Loading';
+import Error from './components/Error/Error';
+import { Suspense, useState } from 'react';
+import { ErrorBoundary } from 'react-error-boundary';
+import List from './components/List/List';
+import Modal from './components/Modal/Modal';
+
+function App() {
+  const [isOpen, setIsOpen] = useState(false);
+  return (
+    <>
+      {!isOpen ? (
+        <button type="button" onClick={() => setIsOpen(!isOpen)}>
+          Show more parameters
+        </button>
+      ) : (
+        ''
+      )}
+      <h1>CO2 emissions data by countries</h1>
+
+      <ErrorBoundary fallback={<Error />}>
+        <Suspense fallback={<Loading />}>
+          <Modal isopen={isOpen} closeOpen={setIsOpen} />
```

Comment:
я прям настоятельно советую избегать использования функции для изменения состояния на прямую речь о setIsOpen 
```suggestion
  const [isOpen, setIsOpen] = useState(false);
  
  const handleOpen = () => {
    setIsOpen(prev => !prev) 
  }
  return (
    <>
      {!isOpen ? (
        <button type="button" onClick={handleOpen}>
          Show more parameters
        </button>
      ) : (
        ''
      )}
      <h1>CO2 emissions data by countries</h1>

      <ErrorBoundary fallback={<Error />}>
        <Suspense fallback={<Loading />}>
          <Modal isopen={isOpen} closeOpen={handleOpen} />
```

такой подход позволяет в одном месте управлять открытием или закрытием так же дает возможность при необходимости оптимизировать путем обертования в useCallBack.  Понимаю что кажется это избыточным но просто прийми как данность